### PR TITLE
Switch mostly to robolectric tests

### PR DESCRIPTION
**Pull request by yschimke**
#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -3,8 +3,6 @@
     <option name="ENABLE_SECOND_REFORMAT" value="true" />
     <option name="SOFT_MARGINS" value="100" />
     <JetCodeStyleSettings>
-      <option name="NAME_COUNT_TO_USE_STAR_IMPORT" value="2147483647" />
-      <option name="NAME_COUNT_TO_USE_STAR_IMPORT_FOR_MEMBERS" value="2147483647" />
```
Comment (luizgrp) [inline .idea/codeStyles/Project.xml:7]: is this changes coming from a stable studio? otherwise you might not want to merge it

**Comment by yschimke** [comment]
There is one I had to disable, and will raise a robolectric bug.  Otherwise it seems no worse than what we have with intrusmentation tests.

File: compose-layout/src/test/java/com/google/android/horologist/compose/layout/FillMaxRectangleTest.kt [hunk 2]
```diff
@@ -31,20 +28,26 @@ import androidx.test.filters.MediumTest
...
 @MediumTest
+@RunWith(RobolectricTestRunner::class)
 class FillMaxRectangleTest {
     @get:Rule
     val composeTestRule = createComposeRule()
 
     @Test
+    @Config(
+        sdk = [30],
+        qualifiers = "w227dp-h227dp-small-notlong-notround-watch-xhdpi-keyshidden-nonav"
```
Comment (luizgrp) [inline compose-layout/src/test/java/com/google/android/horologist/compose/layout/FillMaxRectangleTest.kt:46]: haven't put my mind into this https://github.com/google/horologist/pull/1178#discussion_r1159486525 yet, any plans to implement it in a future pr?

File: compose-layout/src/test/java/com/google/android/horologist/compose/navscaffold/NavScaffoldTest.kt [hunk 3]
```diff
@@ -61,17 +61,27 @@ import kotlinx.coroutines.ExperimentalCoroutinesApi
...
+    qualifiers = "w227dp-h227dp-small-notlong-round-watch-xhdpi-keyshidden-nonav"
+)
 class NavScaffoldTest {
     @get:Rule
     val composeTestRule = createComposeRule()
 
     lateinit var navController: NavHostController
 
     @Suppress("DEPRECATION")
+    @Ignore("Failing ith robolectric")
```
Comment (luizgrp) [inline compose-layout/src/test/java/com/google/android/horologist/compose/navscaffold/NavScaffoldTest.kt:84]: are you going to raise an issue to track these failing tests?

File: media-ui/src/androidTest/java/com/google/android/horologist/media/ui/components/MediaControlButtonsTest.kt [hunk 4]
```diff
@@ -210,7 +211,7 @@ class MediaControlButtonsTest {
         }
 
         // then
-        composeTestRule.onNode(hasProgressBar())
+        composeTestRule.onNode(com.google.android.horologist.test.toolbox.testdoubles.hasProgressBar())
```
Comment (luizgrp) [inline media-ui/src/androidTest/java/com/google/android/horologist/media/ui/components/MediaControlButtonsTest.kt:214]: nit: java fully qualified function name here and in the 5 classes below

**Review by luizgrp — APPROVED** [review]
any particular reason for the switch?

File: compose-layout/src/test/java/com/google/android/horologist/compose/navscaffold/NavScaffoldTest.kt (hunk 3 above)
Comment (yschimke) [reply compose-layout/src/test/java/com/google/android/horologist/compose/navscaffold/NavScaffoldTest.kt:84]: Yep.

File: compose-layout/src/test/java/com/google/android/horologist/compose/layout/FillMaxRectangleTest.kt (hunk 2 above)
Comment (yschimke) [reply compose-layout/src/test/java/com/google/android/horologist/compose/layout/FillMaxRectangleTest.kt:46]: I can follow up.

Comment (yschimke) [reply compose-layout/src/test/java/com/google/android/horologist/compose/layout/FillMaxRectangleTest.kt:46]: https://github.com/google/horologist/issues/1184

File: compose-layout/src/test/java/com/google/android/horologist/compose/navscaffold/NavScaffoldTest.kt (hunk 3 above)
Comment (yschimke) [reply compose-layout/src/test/java/com/google/android/horologist/compose/navscaffold/NavScaffoldTest.kt:84]: https://github.com/google/horologist/issues/1185

**Comment by yschimke** [comment]
> any particular reason for the switch?

They were just too flaky.  The ROI was negative.

I'm hoping we can implement much better tests on Robolectric, more complete, more edgecases.

And upstream bugs or fixes to robolectric.

File: audio/src/test/java/com/google/android/horologist/audio/SystemAudioRepositoryTest.kt [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+            repository.close()
+
+            assertThat(repository.audioOutput.value).isEqualTo(AudioOutput.None)
+            assertThat(repository.available.value).isEmpty()
+        }
+    }
+
+    @Test
+    @Ignore("File robolectric bug for MediaRouter")
```
Comment (yschimke) [inline audio/src/test/java/com/google/android/horologist/audio/SystemAudioRepositoryTest.kt:77]: https://github.com/robolectric/robolectric/issues/8130

cc @luizgrp

**Pull request merged**